### PR TITLE
fix: Improve set attributes BinningX and BinningY

### PR DESCRIPTION
**Pull request by DeboraTahara**
Depends On:
- [ ] https://github.com/tidewise/tidewise.common-package_set/pull/172

### CODE REVIEW

File: src/CameraAravis.cpp [hunk 1]
```diff
@@ -276,21 +276,21 @@ namespace camera
 				arv_camera_set_exposure_time_auto(camera, ARV_AUTO_OFF);				
 				arv_camera_set_exposure_time(camera, value);
 				double exposure = arv_camera_get_exposure_time(camera);
-				if (exposure != value) throw runtime_error("The attribute 'exposure' could not be set");		
+				if (abs(exposure - value) > 50) throw runtime_error("The attribute 'exposure' could not be set "  + std::to_string(value) + " but the camera reports " + std::to_string(exposure));
```
Comment (doudou) [inline src/CameraAravis.cpp:279]: I don't think it's a good idea. Why would '50' be good ?

I think the right way to handle it is in your camera configuration, that is: set a value that is valid for the camera.

File: src/CameraAravis.cpp [hunk 2]
```diff
@@ -354,26 +352,28 @@ namespace camera
...
-		            arv_camera_stop_acquisition(camera); 
-                            arv_camera_set_frame_rate(camera, value);
-                            double fps = arv_camera_get_frame_rate(camera);
-			    if (fabs(fps - value) > 0.01) throw runtime_error("The attribute 'FrameRate' could not be set");
-			    arv_camera_start_acquisition(camera);
-                            return true;
-		    }
-                    default:
-                            return false;
+			case double_attrib::FrameRate:{
```
Comment (doudou) [inline src/CameraAravis.cpp:357]: This change add tabs. Use only spaces.

File: src/CameraAravis.cpp [hunk 3]
```diff
@@ -354,26 +352,28 @@ namespace camera
...
+				arv_camera_stop_acquisition(camera);
+				arv_camera_set_frame_rate(camera, value);
+				double fps = arv_camera_get_frame_rate(camera);
+				if (fabs(fps - value) > 0.01) {
+					throw runtime_error("The attribute 'FrameRate' could not be set, tried to set to " + std::to_string(value) + " but the camera reports " + std::to_string(fps));
+				}
+				arv_camera_start_acquisition(camera);
```
Comment (doudou) [inline src/CameraAravis.cpp:364]: I don't think this line makes sense. What's the driver ?

File: src/CameraAravisTest.cpp [hunk 4]
```diff
@@ -38,11 +38,20 @@ int main(int argc, const char *argv[])
...
 	cam->setAttrib(camera::enum_attrib::WhitebalModeToAuto);
-	cam->setFrameSettings(base::samples::frame::frame_size_t(), base::samples::frame::MODE_BAYER, 8, false);
+	cam->setAttrib(camera::double_attrib::FrameRate, 24);
+	cam->setAttrib(camera::int_attrib::ExposureValue, 1000);
+	cam->setAttrib(camera::int_attrib::GainValue, 40);
+	// cam->setAttrib(camera::int_attrib::RegionX, 1000);
+	// cam->setAttrib(camera::int_attrib::RegionY, 1000);
+	cam->grab(camera::Continuously, 50);
+	cam->setFrameSettings(base::samples::frame::frame_size_t(1248, 720) ,base::samples::frame::MODE_BAYER, 0, false);
+
```
Comment (doudou) [inline src/CameraAravisTest.cpp:54]: This is as specific to the camera you were testing as the previous version ... I don't think it makes sense to commit your changes.

**Comment by DeboraTahara** [comment]
We are closing this PR, because we decided to implement the sdk from lucid camera than implement camera_aravis driver.

**Pull request closed without merging**